core-web/libs/sdk/react/src/lib/next/components/DotCMSEditableText/DotCMSEditableText.tsx

```diff
@@ -0,0 +1,173 @@
+import { Editor } from '@tinymce/tinymce-react';
+import { useEffect, useRef, useState } from 'react';
+
+import { sendMessageToUVE, getUVEState } from '@dotcms/uve';
+import { __DOTCMS_UVE_EVENT__, __TINYMCE_PATH_ON_DOTCMS__ } from '@dotcms/uve/internal';
+import { DotCMSUVEAction, UVE_MODE } from '@dotcms/uve/types';
+
+import { DotCMSEditableTextProps, TINYMCE_CONFIG } from './utils';
+
+/**
+ * Allows inline edit content pulled from dotCMS API using TinyMCE editor
+ *
+ * @export
+ * @component
+ * @param {Readonly<DotCMSEditableTextProps>} props {
+ *     mode = 'plain',
+ *     format = 'text',
+ *     contentlet,
+ *     fieldName = ''
+ * }
+ * @example
+ * ```javascript
+ * import { DotCMSEditableText } from '@dotcms/react';
+ *
+ * const MyContentletWithTitle = ({ contentlet }) => (
+ *     <h2>
+ *         <DotCMSEditableText
+ *             contentlet={contentlet}
+ *             fieldName="title"
+ *             mode='full'
+ *             format='text'/>
+ *     </h2>
+ * );
+ * ```
+ * @returns {JSX.Element} A component to edit content inline
+ */
+export function DotCMSEditableText({
+    mode = 'plain',
+    format = 'text',
+    contentlet,
+    fieldName = ''
+}: Readonly<DotCMSEditableTextProps>): JSX.Element {
+    const editorRef = useRef<Editor['editor'] | null>(null);
+    const [scriptSrc, setScriptSrc] = useState('');
+    const [isInsideEditor, setIsInsideEditor] = useState(false);
+    const [content, setContent] = useState(contentlet?.[fieldName] || '');
+
+    useEffect(() => {
+        setIsInsideEditor(getUVEState()?.mode === UVE_MODE.EDIT);
+
+        if (!contentlet || !fieldName) {
+            console.error('DotCMSEditableText: contentlet or fieldName is missing');
+            console.error('Ensure that all needed props are passed to view and edit the content');
+
+            return;
+        }
+
+        if (getUVEState()?.mode !== UVE_MODE.EDIT) {
+            console.warn('DotCMSEditableText: TinyMCE is not available in the current mode');
```

Comment:
Is this warning needed?

Reply:
I added it for us to debug in case something goes wrong. I will add a check to know that we are on the UVE, so it does not pollute users console on production